app/src/functions/notify-endpoint-usage/app.ts

```diff
@@ -55,17 +57,42 @@ function getRelaysUsed(networkData: QueryAppResponse[], influxData: GetUsageData
   return applicationsData
 }
 
-exports.handler = async () => {
-  const usage = await getUsageData();
+async function getNetworkData(influxData: GetUsageDataQuery[]): Promise<QueryAppResponse[]> {
+  const sleep = (seconds: number, factor: number) => new Promise(resolve => setTimeout(resolve, (seconds ** factor) * 1000))
+
+  const networkApps: QueryAppResponse[] = []
+  const failedApps: GetUsageDataQuery[] = []
 
-  const networkApps = Promise.allSettled(
-    usage.map((app) => getApplicationNetworkData(app.applicationPublicKey))
-  );
+  for (let i = 0; i < maxRetries; i++) {
+    if (i > 0) {
+      await sleep(2, i)
+    }
+
+    const appsToQuery = i === 0 ? influxData : failedApps
+    const networkResponse = await Promise.allSettled(
+      appsToQuery.map((app) => getApplicationNetworkData(app.applicationPublicKey))
+    );
+
+    networkResponse.forEach((app, idx) => {
+      if (app.status === 'fulfilled' && app.value !== undefined) {
```

Comment:
This `status`, where does it come from? I actually think this comes from the Promises themselves haha

Reply:
yeah it does come from there